### PR TITLE
mvcc: reduce value clone in prewrite

### PR DESCRIPTION
**Pull request by Connor1996**
## What have you changed? (mandatory)
take `value` from `mutation` instead of clone

## What are the type of the changes? (mandatory)
- Improvement (non-breaking change which is an improvement to an existing feature)

## How has this PR been tested? (mandatory)
No need

## Does this PR affect documentation (docs/docs-cn) update? (mandatory)
No

## Does this PR affect tidb-ansible update? (mandatory)
No

### CODE REVIEW

File: src/storage/mvcc/txn.rs [hunk 1]
```diff
@@ -132,62 +132,63 @@ impl<S: Snapshot> MvccTxn<S> {
...
 
-        self.lock_key(
-            key.clone(),
-            LockType::from_mutation(&mutation),
-            primary.to_vec(),
-            options.lock_ttl,
-            short_value,
-        );
+        let (key, value) = match mutation {
+            Mutation::Put((key, value)) => (key, Some(value)),
```
[the remark this answers is not in the document]
Comment (Connor1996) [reply src/storage/mvcc/txn.rs]: if that will cause lifetime issue that `self.put_value()` can not use a moved `value`

[the remark this answers is not in the document]
Comment (breeswish) [reply src/storage/mvcc/txn.rs]: wow. interesting!

**Comment by siddontang** [comment]
em, can you show me where to reduce the value clone?

**Comment by Connor1996** [comment]
@siddontang the original L171 and L190

**Comment by siddontang** [comment]
The origin L171 handles short value, but origin L190 handles not short value, seem not the same? @Connor1996

**Comment by Connor1996** [comment]
@siddontang  so it makes if-else in L179

**Comment by siddontang** [comment]
LGTM

**Pull request merged**